src/api/libcellml/component.h

```diff
@@ -435,8 +438,10 @@ class LIBCELLML_EXPORT Component: public ComponentEntity, public ImportedEntity
 
     bool doEquals(const EntityPtr &other) const override; /**< Virtual implementation method for equals, @private. */
 
-    struct ComponentImpl; /**< Forward declaration for pImpl idiom, @private. */
-    ComponentImpl *mPimpl; /**< Private member to implementation pointer, @private. */
+    class ComponentImpl; /**< Forward declaration for pImpl idiom, @private. */
+
+    ComponentImpl *pFunc(); /**< Getter for private implementation pointer, @private. */
+    ComponentImpl const *pFunc() const; /**< Const getter for private implementation pointer, @private. */
```

Comment:
Why do we now have `pFunc()` and can't just use `mPimpl` as we used to? Is that because we have a "normal" getter and a `const` one? If so, where do we need the `const` one?

Otherwise, for consistency, we might want to have:
```c++
const ComponentImpl *pFunc() const;
```
rather than:
```c++
ComponentImpl const *pFunc() const;
```

Reply:
It is there to enforce `const` correctness` which we didn't have before (naughty us).  It gets used in any `const` method.

Reply:
28bd6d2